src/DependencyInjection/ShopwareAppExtension.php

```diff
@@ -57,5 +59,25 @@ public function load(array $configs, ContainerBuilder $container): void
             ->replaceArgument(0, $config['name'])
             ->replaceArgument(1, $config['secret'])
             ->replaceArgument(2, $config['confirmation_url']);
+
+        $this->registerListener($container, $config);
+    }
+
+    /**
+     * @param array<string, bool> $config
+     */
+    private function registerListener(ContainerBuilder $container, array $config): void
+    {
+        $container->setParameter(
+            sprintf('%s.check_if_shop_url_is_reachable', Configuration::EXTENSION_ALIAS),
+            $config['check_if_shop_url_is_reachable']
+        );
+
+        $listener = new Definition(BeforeRegistrationStartsListener::class);
+        $listener->setArgument(0, new Reference(HttpClientInterface::class));
+        $listener->setArgument(1, $config['check_if_shop_url_is_reachable']);
+        $listener->addTag('kernel.event_listener');
+
+        $container->setDefinition(BeforeRegistrationStartsListener::class, $listener);
```

Comment:
Actually, I think I don't need this code because maybe is enough to define the parameter and define the service in the services.xml

WDYT @shyim ?